hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/health/BaselineProvider.java

```diff
@@ -0,0 +1,15 @@
+package org.hypertrace.graphql.entity.health;
+
+import io.reactivex.rxjava3.core.Single;
+import org.hypertrace.core.graphql.context.GraphQlRequestContext;
+import org.hypertrace.gateway.service.v1.baseline.BaselineEntitiesRequest;
+import org.hypertrace.gateway.service.v1.baseline.BaselineEntitiesResponse;
+import org.hypertrace.gateway.service.v1.entity.EntitiesRequest;
+import org.hypertrace.gateway.service.v1.entity.EntitiesResponse;
+import org.hypertrace.graphql.entity.request.EntityRequest;
+
+public interface BaselineProvider {
+    Single<BaselineEntitiesRequest> buildRequest(EntitiesRequest entitiesRequest, EntitiesResponse entitiesResponse, EntityRequest request);
```

Comment:
Any benefit of separating these? build + make requests when they're implemented together?

Reply:
Separate methods is fine and good from an implementation standpoint - but not from an API. Is there any reason to expose an api that needs two steps just to break up the implementation?